src/core/config/openthread-core-config-check.h

```diff
@@ -42,6 +42,10 @@
 #error "OPENTHREAD_ENABLE_DHCP6_MULTICAST_SOLICIT requires OPENTHREAD_CONFIG_DHCP6_CLIENT_ENABLE to be also set."
 #endif
 
+#if OPENTHREAD_ENABLE_DHCP6_MULTICAST_SOLICIT && OPENTHREAD_CONFIG_DHCP6_SERVER_ENABLE
```

Comment:
Is the check appropriate? `OPENTHREAD_ENABLE_DHCP6_MULTICAST_SOLICIT` is primarily for the client side implementation.

Reply:
We thought it would be incorrect to have both enabled at the same time. This was a way to catch our customers to remind them to enable dnsmasq on the border router. That is admittedly a special case for our examples.

Reply:
Makes sense. I am OK to keep this check.